kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableWriteSuiteBase.scala

```diff
@@ -314,6 +315,10 @@ trait DeltaTableWriteSuiteBase extends AnyFunSuite with TestUtils {
       txnBuilder = txnBuilder.withTableProperties(engine, tableProperties.asJava)
     }
 
+    if (withDomainMetadataSupported) {
+      txnBuilder.withDomainMetadataSupported()
```

Comment:
```suggestion
      txnBuilder = txnBuilder.withDomainMetadataSupported()
```

Reply:
err @huan233usc any idea why the tests didn't fail without this change?

Reply:
https://github.com/delta-io/delta/blob/2011ffd5bc8bf6345de3472d85afd58a2ec8051b/kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java#L122
withDomainMetadataSupported() actually update itself.

Reply:
ah yeah, our builder is kind of silly :)